src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -385,35 +387,7 @@ private static List<Class<?>> computeUniqueCatchList(Class<?>[] exceptions) {
             // This exception is unique (so far): add it to the list to catch.
             uniqueList.add(ex);
         }
-        return uniqueList;
-    }
-
-    /**
-     * Convert a fully qualified class name that uses '.' as the package
-     * separator, the external representation used by the Java language
-     * and APIs, to a fully qualified class name that uses '/' as the
-     * package separator, the representation used in the class file
-     * format (see JVMS section {@jvms 4.2}).
-     */
-    private static String dotToSlash(String name) {
-        return name.replace('.', '/');
-    }
-
-    /**
-     * Return the number of abstract "words", or consecutive local variable
-     * indexes, required to contain a value of the given type.  See JVMS
-     * section {@jvms 3.6.1}.
-     * <p>
-     * Note that the original version of the JVMS contained a definition of
-     * this abstract notion of a "word" in section 3.4, but that definition
-     * was removed for the second edition.
-     */
-    private static int getWordsPerType(Class<?> type) {
-        if (type == long.class || type == double.class) {
-            return 2;
-        } else {
-            return 1;
-        }
+        return uniqueList.stream().map(ex -> ClassDesc.ofDescriptor(ex.descriptorString())).toList();
```

Comment:
It would be useful to add a helper method to convert Class to ClassDesc:

     private static ClassDesc toClassDesc(Class<?> type) {
         return ClassDesc.ofDescriptor(type.descriptorString());
     }
     
```suggestion
        return uniqueList.stream().map(ProxyGenerator::toClassDesc).toList();
```